provider/cloudflare.go

```diff
@@ -278,16 +281,20 @@ func newCloudFlareChanges(action string, endpoints []*endpoint.Endpoint, proxied
 }
 
 func newCloudFlareChange(action string, endpoint *endpoint.Endpoint, proxied bool) *cloudFlareChange {
+	ttl := defaultCloudFlareRecordTTL
 	if proxied && (cloudFlareTypeNotSupported[endpoint.RecordType] || strings.Contains(endpoint.DNSName, "*")) {
 		proxied = false
 	}
+	//// min value:120
+	if endpoint.RecordTTL.IsConfigured() && endpoint.RecordTTL >= 120 {
```

Comment:
I am not sure if this logic belongs in External-DNS. I don't think we should try to do validation on behalf of DNS provider. This value might change and we don't want to keep track of it. If a user specifies a value under 120, then we should provide reasonable error log message and ask user to fix it.

Reply:
Yes I agree, usually I would expect this to be handled by the DNS provider. However if you would not implement it and you would set the TTL below 120 it will fail:

`ERRO[0302] failed to create record: error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":1004,\"message\":\"DNS Validation Error\",\"error_chain\":[{\"code\":9021,\"message\":\"Invalid TTL. Must be between 120 and 2,147,483,647 seconds, or 1 for automatic\"}]}],\"messages\":[],\"result\":null}"`

I can remove the `&& endpoint.RecordTTL >= 120` part and document it in **docs/tutorials/cloudflare.md** to make people aware of it.

WDYT?

Reply:
Yeah i think having a documentation plus handling the error with proper log message would be enough in my opinion.